src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -162,7 +193,7 @@ internal void RenderRootVisual(SKSurface surface, Vector2? offsetOverride = null
 	/// Position a sub visual on the canvas and draw its content.
 	/// </summary>
 	/// <param name="parentSession">The drawing session of the <see cref="Parent"/> visual.</param>
-	private void Render(in PaintingSession parentSession)
+	private void Render(in PaintingSession parentSession, bool isFlyoutSurface)
```

Comment:
Should this flag be part of the PaintingSession?

Reply:
For now, probably not. Right now, all the "flyout surface" logic is only visible in Visual. Until we have a very clear reason for why a subclass would need to know about whether the current drawing session is for flyouts or not, let's keep the "layering" details in Visual. This follows the recent changes that pack all the rendering (not painting) logic in Visual directly.

Reply:
I'm not sure though if this can introduce perf regression given how deep the recursion can go. I *think* the whole idea of having `PaintingSession` passed as `in` is to avoid copying individual fields. But introducing a parameter this way will have an extra copy, which kind-of defeats the purpose.

Reply:
I'm not particularly convinced that this would cause any perf regressions. Let's keep it this way until we have a clear example where it does.

Reply:
I agree with moving this to the PaintingSession. There's no point in adding additional context here, particularly when it's created at the same time as the PaintingSession instance.

Also, shouldn't this be using popup-related wording? A flyout is a popup.